vantage6/server/resource/user.py

```diff
@@ -341,6 +350,15 @@ def patch(self, id):
                 if denied:
                     return denied, HTTPStatus.UNAUTHORIZED
 
+            # validate that the assigned role is either a general role or a
+            # role pertaining to that organization
+            if (role.organization and
+                    role.organization.id != user.organization_id):
```

Comment:
Cool that this works <:o. I expected `role.organization.id` to raise an error. But it will first evaluate `role.organization`..